Updated steps necessary for deactivating/deleting a compute node (SCRD-8470)

guglo opened the pull request:
This commit includes:

1. Add steps when delete a  compute.
2. Make sure actions are not available for non compute nodes when
operation is in progress in the Server Summary.
3. Move up the step of disabling compute service in deactivate a compute
4. Handle instances evacuation if the compute node is not reachable when deactivate a compute. 
5. Moved some action names to the constants.js

Commit:
Added steps necessary for delete compute (SCRD-8470)

This commit includes:

1. Add steps when delete a  compute.
2. Make sure actions are not available for non compute nodes when
operation is in progress.
3. Move up the step of disabling compute service in deactivate a compute

Commit:
Updated to deal with not reachable compute node when deactivate

eapodaca reviewed (approved):
This looks mostly OK. I have not tested.

GarySmith commented on src/localization/bundles/en.json:
```diff
@@ -686,14 +686,16 @@
     "server.migrate_title": "Migrate instances from {0}",
     "server.migrate.prompt": "Select a computer host to migrate the instances on {0} host to.",
-    "server.deactivate.confirm.message_instances_cant_be_migrated": "Are you sure you want to deactivate this compute server ({0})? It is currently hosting {1} instance(s) which may be terminated for made in accessible by this action.",
+    "server.evacuate.prompt": "Select a computer host to evacuate the instances on {0} host to.",
```
I believe this is supposed to say "compute host" instead of "computer host"

GarySmith commented on src/localization/bundles/en.json:
```diff
@@ -686,14 +686,16 @@
     "server.migrate.prompt": "Select a computer host to migrate the instances on {0} host to.",
-    "server.deactivate.confirm.message_instances_cant_be_migrated": "Are you sure you want to deactivate this compute server ({0})? It is currently hosting {1} instance(s) which may be terminated for made in accessible by this action.",
+    "server.evacuate.prompt": "Select a computer host to evacuate the instances on {0} host to.",
+    "server.deactivate.confirm.message_instances_terminate": "Are you sure you want to deactivate this compute server ({0})? It is currently hosting {1} instance(s) which may be terminated for made in accessible by this action.",
```
"for made" is supposed to be "or made"

GarySmith commented on src/pages/ManageComputeHost/DeactivateComputeHost.js:
```diff
@@ -55,17 +48,17 @@ class DeactivateComputeHost extends DisableComputeServiceNetwork {
+      action: ::this.disableCompServices
+    }];
+
+    let instancesPlaybooks = this.getPlaybooksForInstances();
```
Ditto above, these 4 lines should just be `playbooks = playbooks.concat(this.getPlaybooksForInstances());`

GarySmith commented on src/pages/ReplaceServer/DisableComputeServiceNetwork.js:
```diff
@@ -363,34 +351,39 @@ class DisableComputeServiceNetwork extends BaseUpdateWizardPage {
+      playbooks: [constants.REMOVE_FROM_AGGREGATES_ACTION]
+    }];
+
+    let instancesSteps = this.getStepsForInstances();
```
Ditto

GarySmith commented on src/pages/ManageComputeHost/DeleteComputeHost.js:
```diff
@@ -14,9 +14,143 @@
+          this.props.operationProps.oldServer.hostname;
+        logProgressResponse(logger, response, logMsg);
+      })
+      .catch((error) => {
```
This error handling logic is pretty convoluted, and seems to be copy/pasted below. It's unclear why a failure dialog is only presented for some 500 errors and not others. Or why some messages are translated and others are not, or why some errors cause an error to be thrown and others do not.

guglo replied:
@GarySmith  it is copied and pasted. I tried to move them to a common place like util, but gave it up , the effort seems to be more involved than I initially thought. 
When delete aggregates, since the backend deletes host aggregates one by one, if there are errors, we don't stop, in the end if the results include good and bad responses, backends returns response with 500 status code. Then UI checks if error code is 500 and have failed (bad) and have deleted(good), pops a confirm dialog to ask user to continue or not.

guglo replied:
Talked to Gary, I will add some comments so we will be a little bit clear about the error handling part.

GarySmith commented on src/pages/ReplaceServer/DisableComputeServiceNetwork.js:
```diff
@@ -406,10 +399,31 @@ class DisableComputeServiceNetwork extends BaseUpdateWizardPage {
+      })
+    }];
+
+    let instancesPlaybooks = this.getPlaybooksForInstances();
```
Ditto

GarySmith reviewed (commented):
Lots of small stuff

Commit:
Updated to simplify the code and fix messages

dotNomad reviewed (approved):
Tested and seems to be working as intended. Both steps (removing aggregates and disabling network agents) appear when deleting a disabled compute node.

jeremy-moffitt commented:
> Tested and seems to be working as intended. Both steps (removing aggregates and disabling network agents) appear when deleting a disabled compute node.

How did you simulate the delete? When I try deactivating in a dev environment the node just goes back to active since there's not a real backend behind it

Commit:
Added some comments in the  error handlings.

guglo commented:
@jeremy-moffitt in the ardana-service tools/compute-mock-data.json "compute_services_status": {
    "nova-compute": false
  },
refresh delete will show up
if you want to test evacuate step, add some in get_instances
"get_instances": [{
    "id": "8279e65d-6e87-4a50-b789-96edd753fbb2",
    "name": "test3"
  }, {
    "id": "1d51f18f-27fd-4c34-a0aa-c07a5e9462e7",
    "name": "test2"
  }]

jeremy-moffitt reviewed (approved):
tried it out with the suggested tweaks to make the compute servers deletable, seemed fine... not merging though pending confirmation from @GarySmith that his concerns have been addressed

Commit:
Updated message and fixed the issue when there are no instances

jeremy-moffitt reviewed (approved):
re-tested with the latest change, still seems to work ok. One thing I found a little odd, the first time you open the delete dialog, there is a comparatively lengthy "loading please wait..." that I think was checking about a password or something (the password prompt is what shows once it loads). The timing is a bit odd, but it works

Commit:
Fixed a couple of issues from testing on real system

Updated messages
check reachable before run nova stop in deactivate compute
check compute nodes size not 0 when run cobbler deploy

jeremy-moffitt commented on src/pages/UpdateServers.js:
```diff
@@ -688,7 +688,7 @@ class UpdateServers extends BaseUpdateWizardPage {
           </If>
           <If condition={!loading && server.hasInstances && otherHosts.length === 0}>
-            {translate('server.deactivate.confirm.message_instances_terminate', id, instances.length)}
+            {translate('server.deactivate.confirm.message_instances_terminate', id, instances.length, id)}
```
couldnt you just reuse {0} or will the string replacement not work that way?

Commit:
Updated message to remove extra paremeter

jeremy-moffitt reviewed (approved):
thanks for fixing the repeat parameter issue!
